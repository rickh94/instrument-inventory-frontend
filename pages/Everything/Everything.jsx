import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { LoadingHeader, InstrumentTable } from '../../components'
import { Paper } from '@material-ui/core'
import {makeStyles} from '@material-ui/styles'
import { API } from 'aws-amplify'
import { List as TableLoader } from 'react-content-loader'

const useStyles = makeStyles({
  root: {
    // margin: '0 10px',
    // paddingRight: 10,
    paddingLeft: 10,
    paddingTop: 10,
    paddingBottom: 0,
    overflow: 'hidden',
    height: '90vh',
    width: '100%',
    // width: '98vw',
  },
  tableWrapper: {
    height: '95%',
    width: '100%',
    margin: 0,
    padding: 0,
    overflow: 'hidden',
  },
})

const Everything = ({ showAlert }) => {
  const [isLoading, setLoading] = useState(false)
  const [records, setRecords] = useState([])
  useEffect(() => {
    const getRecords = async () => {
      setLoading(true)
      try {
        const res = await API.get('instrument-inventory', 'instruments/all')
        setRecords(res)
      } catch (err) {
        showAlert(err)
      }
      setLoading(false)
    }

    getRecords()
  }, [])

  const classes = useStyles()
  return (
    <React.Fragment>
      <div className={classes.root}>
        <LoadingHeader isLoading={isLoading} title="Everything" />
        {isLoading ? (
          <TableLoader />
        ) : (
          <div className={classes.tableWrapper}>
            <InstrumentTable records={records} />
          </div>
        )}
      </div>
    </React.Fragment>
  )
}

Everything.propTypes = {
  showAlert: PropTypes.func.isRequired,
}

export default Everything
