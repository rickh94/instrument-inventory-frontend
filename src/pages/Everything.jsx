import React, { useState, useEffect, useContext } from 'react'
import { LoadingHeader, InstrumentTable } from '../components'
import { makeStyles } from '@material-ui/styles'
import { API } from 'aws-amplify'
import { List as TableLoader } from 'react-content-loader'
import { HelpersContext } from '../contexts'
import Grid from '@material-ui/core/Grid'
import { Button } from '@material-ui/core'
import Papa from 'papaparse'

const useStyles = makeStyles({
  root: {
    paddingLeft: 10,
    paddingTop: 10,
    paddingBottom: 0,
    overflow: 'hidden',
    height: '90vh',
    width: '100%',
  },
  tableWrapper: {
    height: '95%',
    width: '100%',
    margin: 0,
    padding: 0,
    overflow: 'hidden',
  },
})

const Everything = () => {
  const { showAlert } = useContext(HelpersContext)
  const [isLoading, setLoading] = useState(false)
  const [records, setRecords] = useState([])
  useEffect(() => {
    const getRecords = async () => {
      setLoading(true)
      try {
        const res = await API.get('instrument-inventory', 'instruments/all', {})
        console.log(res)
        setRecords(res.instruments)
        if (res.instrumentsFailed.length > 0) {
          console.error(res.instrumentsFailed)
        }
      } catch (err) {
        showAlert(err.toString())
      }
      setLoading(false)
    }

    getRecords()
  }, [])

  const handleDownloadCsv = () => {
    const csvData = Papa.unparse(records, {
      skipEmptyLines: true,
      header: true,
    })
    const csvDownload = document.createElement('a')
    csvDownload.setAttribute('href', 'data:text/csv;charset=utf-8,' + encodeURIComponent(csvData))
    csvDownload.setAttribute('download', 'instruments-' + (new Date()).toLocaleString() + '.csv')

    csvDownload.click()
  }
  const classes = useStyles()
  return (
    <React.Fragment>
      <div className={classes.root}>
        <Grid container direction="row" justify="space-between">
          <LoadingHeader isLoading={isLoading} title="Everything" />
          <Button variant="contained" color="primary" style={{ marginLeft: '1rem' }} onClick={handleDownloadCsv}>Download
            CSV</Button>
        </Grid>
        {isLoading ? (
          <TableLoader />
        ) : (
          <div className={classes.tableWrapper}>
            <InstrumentTable records={records} />
          </div>
        )}
      </div>
    </React.Fragment>
  )
}

Everything.propTypes = {}

export default Everything
