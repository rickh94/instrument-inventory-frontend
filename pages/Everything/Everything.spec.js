import React from 'react'
import Everything from './Everything'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { API } from 'aws-amplify'

const flushPromises  = new Promise(setImmediate)

const mockInstruments = [
  {
    id: 1,
    number: '1-001',
    type: 'Violin',
    size: '4/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 2,
    number: '4-002',
    type: 'Violin',
    size: '1/4',
    location: 'Storage',
    condition: 4,
    quality: 3,
    gifted: true,
  },
  {
    id: 3,
    number: '3-003',
    type: 'Violin',
    size: '3/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 4,
    number: '2-004',
    type: 'Violin',
    size: '1/2',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 5,
    number: 'C1-001',
    type: 'Cello',
    size: '4/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 6,
    number: 'C1-002',
    type: 'Cello',
    size: '4/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 7,
    number: 'C2-003',
    type: 'Cello',
    size: '1/2',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
  {
    id: 8,
    number: 'C3-004',
    type: 'Cello',
    size: '3/4',
    assignedTo: 'Test Name',
    location: 'Office',
    condition: 4,
    quality: 3,
    gifted: false,
    history: ['one', 'two', 'three'],
  },
]

afterEach(cleanup)

describe('<Everything />', () => {
  it('matches snapshot', () => {
    API.get = jest.fn().mockImplementation(() => Promise.resolve(mockInstruments))
    const { container } = render(
      <MemoryRouter>
        <Everything showAlert={jest.fn()} />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
