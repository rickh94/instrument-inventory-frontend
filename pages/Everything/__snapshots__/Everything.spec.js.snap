// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Everything /> matches snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-heading-3"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Everything
      </h5>
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate makeStyles-loader-4"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
    <svg
      aria-label="Loading interface..."
      preserveAspectRatio="none"
      role="img"
      viewBox="0 0 400 130"
    >
      <title>
        Loading interface...
      </title>
      <rect
        clip-path="url(#2xzxnxrzktn)"
        height="130"
        style="fill: url(#14pwyyg7xgi);"
        width="400"
        x="0"
        y="0"
      />
      <defs>
        <clippath
          id="2xzxnxrzktn"
        >
          <rect
            height="10"
            rx="3"
            ry="3"
            width="250"
            x="0"
            y="0"
          />
          <rect
            height="10"
            rx="3"
            ry="3"
            width="220"
            x="20"
            y="20"
          />
          <rect
            height="10"
            rx="3"
            ry="3"
            width="170"
            x="20"
            y="40"
          />
          <rect
            height="10"
            rx="3"
            ry="3"
            width="250"
            x="0"
            y="60"
          />
          <rect
            height="10"
            rx="3"
            ry="3"
            width="200"
            x="20"
            y="80"
          />
          <rect
            height="10"
            rx="3"
            ry="3"
            width="80"
            x="20"
            y="100"
          />
        </clippath>
        <lineargradient
          id="14pwyyg7xgi"
        >
          <stop
            offset="0%"
            stop-color="#f0f0f0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="#e0e0e0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#f0f0f0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
  </div>
</div>
`;
